src/echo-common/src/poe-zone-tracker-service.ts

```diff
@@ -0,0 +1,465 @@
+      expeditionEncounter: false,
+      expeditionGwennenEncounter: false,
+      expeditionRogEncounter: false,
+      expeditionTujenEncounter: false,
+      blightEncounter: false
+    }
+  }
+
+  private generatingAreaEventSubscribe() {
+    this.logEvents$
+      .pipe(
+        filter((e) => e.type == 'GeneratingAreaEvent'),
+        map((e) => e as PoeGeneratingAreaEvent)
+      )
+      .subscribe((event) => {
+        if (this.currentZoneDelta === undefined) {
+          //Emit as event
+          throw new Error(
+            'Something went wrong. currentZoneDelta not instansiated correctly in GeneratingAreaEvent'
+          )
+        }
+
+        if (!this.currentZoneDelta.startLoadTime) {
+          //Emit as event
+          throw new Error(
+            'Something went wrong. currentZoneDelta.startLoadTime not set correctly in GeneratingAreaEvent'
+          )
+        }
+
+        this.currentZoneDelta.poeZoneInstance.areaLevel = event.areaLevel
+        this.currentZoneDelta.poeZoneInstance.zoneSeed = event.seed
+        this.currentZoneDelta.poeZoneInstance.areaTag = event.areaTag
+
+        //Determine if currentZoneDelta is a revisit of an old zone
+        //Has server, arealevel, seed, areatag and
+        let zoneHash: string =
+          this.currentZoneDelta.poeZoneInstance.zoneServer +
+          '|' +
+          this.currentZoneDelta.poeZoneInstance.areaTag +
+          '|' +
+          this.currentZoneDelta.poeZoneInstance.areaLevel.toString() +
+          '|' +
+          this.currentZoneDelta.poeZoneInstance.zoneSeed.toString()
+
+        if (
+          event.areaTag.toLowerCase().includes('hideout') ||
+          event.areaTag.toLowerCase().includes('town')
+        ) {
+          this.currentZoneDelta.poeZoneInstance.isZoneHideoutOrTown = true
+          zoneHash = zoneHash + '|' + event.time.getTime().toString()
+        } else {
+          this.currentZoneDelta.poeZoneInstance.isZoneHideoutOrTown = false
+        }
+
+        this.currentZoneDelta.poeZoneInstance.zoneHash = zoneHash
+
+        const existingZone = this.zones.get(zoneHash)
+        if (existingZone) {
+          this.currentZoneDelta.poeZoneInstance = existingZone
+          //TODO IF EXISTING ZONE IS FOUND, DO SOME LOGIC BASED ON DATE TO DETERMINE IF IT IS CLOSE ENOUGH
+          existingZone.zoneDeltas.push(this.currentZoneDelta)
+          this.zones.set(zoneHash, existingZone)
+        } else {
+          this.currentZoneDelta.poeZoneInstance.zoneDeltas.push(this.currentZoneDelta)
+          this.zones.set(zoneHash, this.currentZoneDelta.poeZoneInstance)
+        }
+      })
+  }
+
+  private zoneEntranceEventSubscribe() {
+    this.logEvents$
+      .pipe(
+        filter((e) => e.type == 'ZoneEntranceEvent'),
+        map((e) => e as PoeZoneEntranceEvent)
+      )
+      .subscribe((event) => {
+        if (this.currentZoneDelta === undefined) {
+          //Emit as event
+          throw new Error(
```

Comment:
Throwing errors inside a subscribe like this will make it impossible to catch them anywhere. That might be okay if these are errors that should never happen, but even then since they are uncatchable converting them to just logging warnings/errors probably makes more sense.